lib/cli/cliCompact.js

```diff
@@ -0,0 +1,181 @@
+     */
+    execute(args) {
+        const { callback, showHelp } = this.options;
+        const command = args[0];
+        switch (command) {
+            case 'status':
+                return this.statusCompactModeHost();
+            case 'enable':
+            case 'on':
+                return this.setCompactModeHost(true);
+            case 'disable':
+            case 'off':
+                return this.setCompactModeHost(false);
+            default:
+                if (command && command.includes('.')) {
+                    if (args[1]) {
+                        switch (args[1]) {
+                            case 'status':
+                                return this.statusCompactModeInstance(command);
+                            case 'enable':
+                            case 'on':
+                                return this.setCompactModeInstance(command, true, args[2]);
+                            case 'disable':
+                            case 'off':
+                                return this.setCompactModeInstance(command, false);
+                            case 'group':
+                                return this.setCompactModeInstance(command, undefined, args[2]);
+                        }
+                    }
+                }
+
+                messages.error.unknownCommand('compact', command);
+                showHelp();
+                return void callback(3);
+        }
+    }
+
+    /**
+     * Output status of compact mode for host
+     */
+    statusCompactModeHost() {
+        const { callback } = this.options;
+
+        const compactMode = !!(this.config && this.config.system && this.config.system.compact);
+        console.log('Compact mode for this host is currently ' + (compactMode ? 'enabled' : 'disabled'));
+
+        return void callback();
+    }
+
+    /**
+     * Set compact mode for this host
+     * @param targetState target state to set for host
+     */
+    setCompactModeHost(targetState) {
+        const { callback, dbConnect } = this.options;
+
+        dbConnect((_objects, _states, isOffline) => {
+            const compactMode = !!(this.config && this.config.system && this.config.system.compact);
+            console.log('Compact mode for this host is currently ' + (compactMode ? 'enabled' : 'disabled'));
+
+            if (targetState === compactMode) {
+                return void callback();
+            }
+
+            this.config.system = this.config.system || {};
+            this.config.system.compact = targetState;
+            const configFileName = tools.getConfigFileName();
+            fs.writeFileSync(configFileName, JSON.stringify(this.config, null, 4), 'utf8');
+
+            console.log();
+            console.log('Compact mode for this host changed to ' + (targetState ? 'enabled' : 'disabled'));
+
+            if (!isOffline) {
+                console.log();
+                console.log('Please stop ioBroker for this change to take effect!')
```

Comment:
Wouldn't restarting instead of only stopping be more logical from a UX perspective?

Reply:
yes "restart" is better